internal/resource/config/localidentity/local_identity_identity_profiles_resource.go

```diff
@@ -0,0 +1,917 @@
+						if (internaltypes.IsDefined(profilePagefield)) && (profilePagefield.ValueBool()) {
+							resp.Diagnostics.AddError("Invalid Value for Attribute!", fmt.Sprintln("profile_page_field option for the fields attribute should not be set to 'true' when profile is disabled."))
+						}
+						registrationPageField := fieldElemAttrs.Attributes()["registration_page_field"].(basetypes.BoolValue)
+						if (internaltypes.IsDefined(registrationPageField)) && (!registrationPageField.ValueBool()) {
+							resp.Diagnostics.AddError("Invalid Value for Attribute!", fmt.Sprintln("registration_page_field option is required to be set to 'true' for the fields attribute when registration is the only option enabled."))
+						}
+					}
+				}
+				if !model.ProfileConfig.IsNull() {
+					resp.Diagnostics.AddError("Invalid Attribute Combination!", fmt.Sprintln("profile_config is not allowed when profile is not enabled."))
+				}
+			}
+			if (model.ProfileEnabled.ValueBool()) && (model.ProfileConfig.IsNull()) {
+				resp.Diagnostics.AddError("Invalid Value for Attribute!", fmt.Sprintln("profile_config is required when profile is enabled."))
+			}
+			if !model.RegistrationEnabled.ValueBool() {
+				if internaltypes.IsDefined(model.FieldConfig.Attributes()["fields"]) {
+					for _, fieldElem := range fieldElems {
+						fieldElemAttrs := fieldElem.(basetypes.ObjectValue)
+						registrationPageField := fieldElemAttrs.Attributes()["registration_page_field"].(basetypes.BoolValue)
+						if (internaltypes.IsDefined(registrationPageField)) && (registrationPageField.ValueBool()) {
+							resp.Diagnostics.AddError("Invalid Value for Attribute!", fmt.Sprintln("registration_page_field option for the fields attribute should not be set to 'true' when registration is disabled."))
+						}
+						profilePageField := fieldElemAttrs.Attributes()["profile_page_field"].(basetypes.BoolValue)
+						if (internaltypes.IsDefined(profilePageField)) && (!profilePageField.ValueBool()) {
+							resp.Diagnostics.AddError("Invalid Value for Attribute!", fmt.Sprintln("profile_page_field option is required to be set to 'true' for the fields attribute when profile management is the only option enabled."))
+						}
+					}
+				}
+				if !model.RegistrationConfig.IsNull() {
+					resp.Diagnostics.AddError("Invalid Attribute!", fmt.Sprintln("registration_config is not allowed when registration is not enabled."))
+				}
+			}
+			if (model.RegistrationEnabled.ValueBool()) && (model.RegistrationConfig.IsNull()) {
+				resp.Diagnostics.AddError("Invalid Value for Attribute!", fmt.Sprintln("registration_config is required when registration is enabled."))
+			}
+		}
+	}
+}
+
+func readLocalIdentityIdentityProfilesResponse(ctx context.Context, r *client.LocalIdentityProfile, state *localIdentityIdentityProfilesResourceModel) {
+	state.Id = internaltypes.StringTypeOrNil(r.Id, false)
+	state.Name = types.StringPointerValue(&r.Name)
```

Comment:
You're converting to a pointer and then back to a string here. You can just use `types.StringValue(r.Name)`